### ISSUE TITLE
Cannot create new recipes: TypeError: fields is undefined

### ISSUE THREAD

**Issue by Osmose** [mozilla/normandy#493]
When trying to create a new recipe, I get the following error when trying to select the action type:

```
TypeError: fields is undefined[Learn More]  control.js%20line%204545%20%3E%20eval:92:1
	HeartbeatFields https://localhost:8000/static/bundles/control.js%20line%204545%20%3E%20eval:92:1
	_constructComponentWithoutOwner https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:310:27
	_constructComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:278:16
	mountComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:190:16
	mountComponent https://localhost:8000/static/bundles/control.js%20line%20915%20%3E%20eval:47:18
	updateChildren https://localhost:8000/static/bundles/control.js%20line%206364%20%3E%20eval:122:35
	_reconcilerUpdateChildren https://localhost:8000/static/bundles/control.js%20line%206532%20%3E%20eval:210:11
	_updateChildren https://localhost:8000/static/bundles/control.js%20line%206532%20%3E%20eval:314:26
	updateChildren https://localhost:8000/static/bundles/control.js%20line%206532%20%3E%20eval:301:7
	_updateDOMChildren https://localhost:8000/static/bundles/control.js%20line%206394%20%3E%20eval:941:7
	updateComponent https://localhost:8000/static/bundles/control.js%20line%206394%20%3E%20eval:759:5
	receiveComponent https://localhost:8000/static/bundles/control.js%20line%206394%20%3E%20eval:717:5
	receiveComponent https://localhost:8000/static/bundles/control.js%20line%20915%20%3E%20eval:126:5
	_updateRenderedComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:761:7
	_performComponentUpdate https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:739:5
	updateComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:658:7
	receiveComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:552:5
	receiveComponent https://localhost:8000/static/bundles/control.js%20line%20915%20%3E%20eval:126:5
	_updateRenderedComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:761:7
	_performComponentUpdate https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:739:5
	updateComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:658:7
	receiveComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:552:5
	receiveComponent https://localhost:8000/static/bundles/control.js%20line%20915%20%3E%20eval:126:5
	_updateRenderedComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:761:7
	_performComponentUpdate https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:739:5
	updateComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:658:7
	receiveComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:552:5
	receiveComponent https://localhost:8000/static/bundles/control.js%20line%20915%20%3E%20eval:126:5
	_updateRenderedComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:761:7
	_performComponentUpdate https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:739:5
	updateComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:658:7
	receiveComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:552:5
	receiveComponent https://localhost:8000/static/bundles/control.js%20line%20915%20%3E%20eval:126:5
	_updateRenderedComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:761:7
	_performComponentUpdate https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:739:5
	updateComponent https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:658:7
	performUpdateIfNecessary https://localhost:8000/static/bundles/control.js%20line%206382%20%3E%20eval:566:7
	performUpdateIfNecessary https://localhost:8000/static/bundles/control.js%20line%20915%20%3E%20eval:158:5
	runBatchedUpdates https://localhost:8000/static/bundles/control.js%20line%20471%20%3E%20eval:151:5
	perform https://localhost:8000/static/bundles/control.js%20line%201161%20%3E%20eval:138:13
	perform https://localhost:8000/static/bundles/control.js%20line%201161%20%3E%20eval:138:13
	perform https://localhost:8000/static/bundles/control.js%20line%20471%20%3E%20eval:90:12
	flushBatchedUpdates https://localhost:8000/static/bundles/control.js%20line%20471%20%3E%20eval:173:7
	bound flushBatchedUpdates self-hosted:914:17
	closeAll https://localhost:8000/static/bundles/control.js%20line%201161%20%3E%20eval:204:11
	perform https://localhost:8000/static/bundles/control.js%20line%201161%20%3E%20eval:151:11
	batchedUpdates https://localhost:8000/static/bundles/control.js%20line%206484%20%3E%20eval:63:7
	batchedUpdates https://localhost:8000/static/bundles/control.js%20line%20471%20%3E%20eval:98:3
	dispatchEvent https://localhost:8000/static/bundles/control.js%20line%206502%20%3E%20eval:150:7
	bound dispatchEvent self-hosted:956:17
	receiveMessage
```

**Comment by mythmon** [comment]
I was unable to create a recipe, due to a different JS issue, but when I dug in deeper the problem went away. This is working for me now. Can you still reproduce this @Osmose?

**Comment by Osmose** [comment]
CC @raymak

**Comment by andymikulski** [comment]
Fixed in PR #494, waiting for CI to finish before merging (has passed review by Osmose and QA) into master

**Comment by andymikulski** [comment]
@mythmon @Osmose 

Found the other issue when creating recipes. Upon form submission, redux-form fires its own `handleSubmit` prop, which in turn fires our `onSubmit(values, dispatch, ...)`  function.

Apparently, redux-form does _not_ return empty form fields with the `values` param. What's happening is if you try to create a recipe without `Enabled` checked, the `enabled` property is never assigned to the `recipe` object that is passed to the API for creation. The API requires `enabled` to be passed in - [apparently](https://cloud.githubusercontent.com/assets/2767162/22835441/fe1d96fa-ef75-11e6-8a60-d8ea2c9f5490.png) - so by submitting a `recipe` without `enabled`, the error is thrown.

There are a couple things here that stand out as strange to me. First, I can't recall doing any work on the redux-form business that would change how it handles empty fields, so I'm not sure if this is a recent issue or what. Second, looking at the back end, I don't see any 'required' flags on the field, so I would expect that a request without the `enabled` key would just default it to `false`.

Maybe @rehandalal can chime in with any thoughts? I'm working on a front end fix that ensures all required keys actually have values, though it feels a little sticky:

<img width="963" alt="screen shot 2017-02-10 at 9 45 44 am" src="https://cloud.githubusercontent.com/assets/2767162/22835386/c59f14de-ef75-11e6-8545-24694ab13efe.png">

I don't love this - specifically targeting the `enabled` key to ensure it has the correct default value. Feels weird. But, fwiw, it _does_ fix recipe creation.
